src/main/java/com/flansmod/common/guns/EntityBullet.java

```diff
@@ -681,24 +694,54 @@ else if (motionZ != 0)
-                    MovingObjectPosition raytraceResult = blockHit.raytraceResult;
+                    MovingObjectPosition raytraceResult = blockHit.raytraceResult;                    
+                    Vec3 hitVec = raytraceResult.hitVec;
+                    
                     //If the hit wasn't an entity hit, then it must've been a block hit
                     int xTile = raytraceResult.blockX;
                     int yTile = raytraceResult.blockY;
                     int zTile = raytraceResult.blockZ;
+                    
+                    
                     if (FlansMod.DEBUG)
-                        worldObj.spawnEntityInWorld(new EntityDebugDot(worldObj, new Vector3f(raytraceResult.hitVec.xCoord, raytraceResult.hitVec.yCoord, raytraceResult.hitVec.zCoord), 1000, 0F, 1F, 0F));
+                        worldObj.spawnEntityInWorld(new EntityDebugDot(worldObj, new Vector3f(hitVec.xCoord, hitVec.yCoord, hitVec.zCoord), 1000, 0F, 1F, 0F));
                     
                     Block block = worldObj.getBlock(xTile, yTile, zTile);
-                    Material mat = block.getMaterial();
+                    Material mat = block.getMaterial();             
+                    
+                    if(FlansMod.enableBlockPenetration) {
+                    	for(Block penetrableBlock : FlansMod.penetrableBlocks.keySet()) { 
+                        	if(block != penetrableBlock) continue;
+                        	
+                        	Float[] valuesArray = FlansMod.penetrableBlocks.get(penetrableBlock);
+                        	float metadata = valuesArray[0];
+                        	
+                        	if(metadata != -1 && metadata != worldObj.getBlockMetadata(xTile, yTile, zTile)) continue;
+                        	
+                        	float hardness = valuesArray[1];
+                        	
+                        	penetratingPower-=hardness * (type.blockPenetrationModifier > 0 ? (1F/type.blockPenetrationModifier) : 1F);
+                        	if(penetratingPower < 0) break;
+                        	
+                        	FlansMod.proxy.playBlockBreakSound(xTile, yTile, zTile, block, this.dimension);
+                        	if(valuesArray[2] == 1) worldObj.setBlockToAir(xTile, yTile, zTile);  
+                        	
+                        	//The block was penetrated, so the bullet can keep going
+                        	continue hitLoop;
+                        }
+                    }
+                    
+                    if (type.hitSoundEnable)
+                        PacketPlaySound.sendSoundPacket(posX, posY, posZ, type.hitSoundRange, dimension, type.hitSound, true);
+                    
                     //If the bullet breaks glass, and can do so according to FlansMod, do so.
-                    if (type.breaksGlass && mat == Material.glass) {
+                    if(type.breaksGlass && mat == Material.glass) {
```

Comment:
hm.. could we retire `/teams canBreakGlass`, or rework it to fit in with the new system instead?

Reply:
I was thinking the same, but I would make this part of a general overhaul in the future

Reply:
Okay, sure.